realm/realm-library/src/androidTest/kotlin/io/realm/UUIDTests.kt

```diff
@@ -0,0 +1,435 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import io.realm.annotations.PrimaryKey
+import io.realm.annotations.Required
+import io.realm.exceptions.RealmException
+import io.realm.exceptions.RealmPrimaryKeyConstraintException
+import io.realm.kotlin.createObject
+import io.realm.kotlin.where
+import org.junit.After
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.junit.runner.RunWith
+import java.util.*
+
+open class UUIDPrimaryKeyRequired
+    : RealmObject() {
+    @field:PrimaryKey
+    @field:Required
+    var id: UUID? = null
+    var name: String = ""
+    var anotherId: UUID? = null
+}
+
+open class UUIDPrimaryKeyNotRequired
+    : RealmObject() {
+    @field:PrimaryKey
+    var id: UUID? = null
+    var name: String = ""
+
+}
+
+open class UUIDAndString
+    : RealmObject() {
+    var id: UUID? = null
+    var name: String = ""
+}
+
+open class UUIDRequiredRealmList
+    : RealmObject() {
+    var id: Long = 0
+
+    @field:Required
+    var ids: RealmList<UUID> = RealmList()
+    var name: String = ""
+}
+
+open class UUIDOptionalRealmList
+    : RealmObject() {
+    var id: Long = 0
+
+    var ids: RealmList<UUID> = RealmList()
+    var name: String = ""
+}
+
+@RunWith(AndroidJUnit4::class)
+class UUIDTests {
+    private lateinit var realmConfiguration: RealmConfiguration
+    private lateinit var realm: Realm
+
+    @Rule
+    @JvmField
+    val folder = TemporaryFolder()
```

Comment:
Instead of this we have the `TestRealmConfigurationFactory` rule which makes it easier to create custom configurations that are cleaned up afterwards.